src/main/java/duke/Task.java

```diff
@@ -0,0 +1,62 @@
+package duke;
+
+/**
+ * The abstract Task class representing a task.
+ */
+public abstract class Task {
+    private String description;
+    private boolean isDone;
+    private String icon;
+
+    /**
+     * Constructor for a Task object.
+     *
+     * @param description The description of the task.
+     * @param icon Marker that indicates the type of task.
+     */
+    public Task(String description, String icon) {
+        this.description = description;
+        this.icon = icon;
+        this.isDone = false;
+    }
+
+    /**
+     * A method that gets a marker that indicates if task is complete.
+     *
+     * @return A marker that indicates if task is complete.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " ");
+    }
+
+    /**
+     * A method that marks the task as done.
+     */
+    public void markAsDone() {
+        isDone = true;
+    }
+
+    /** A method that indicates whether the task description contains a keyword.
+     *
+     * @param keyword The keyword to be searched for in the description.
+     * @return Boolean indicating whether the keyword is found in the description.
+     */
+    public boolean contains(String keyword) {
```

Comment:
Should this be something like hasKeyword since it is a boolean?

Reply:
Must have missed this, thanks for pointing it out!